runtime/src/chpl-launcher-common.c

```diff
@@ -563,7 +563,7 @@ char* chpl_get_enviro_keys(char sep)
 
     int keyLen = strstr(environ[i], "=") - environ[i];
     // skip keys that end with _modshare
-    if (keyLen > 8 && strncmp(environ[i] + keyLen - 8, "_modshare", 9) == 0) {
+    if (keyLen > 8 && strncmp(environ[i] + keyLen - 9, "_modshare", 9) == 0) {
```

Comment:
I have very little context here, but are you not ignoring `=` in `FOO=value`?

Reply:
kinda. My logic is that `strstr` returns the index of the `=`, which by subtracting the start of the string gives you then length of the key, ignoring the `=`. Then we see if the last 9 characters of the key (`environ[i] + keyLen - 9`) are `_modshare`

Reply:
I have been looking at my fingers trying to do the math for a while. I think this is good.